app/assets/scripts/components/aa-field-map-legend.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import lineColors from '../utils/line-colors';
+
+const AAFieldMapLegend = React.createClass({
+  displayName: 'AAFieldMapLegend',
+
+  propTypes: {
+    sources: React.PropTypes.array
+  },
+
+  renderLegendElements: function () {
+    // for each source as the source name
+    // as well as a small svg path with the color
+    // matching source in the lineColors config
+    return this.props.sources.map((source, i) => {
+      return (
+        <li key={i}>
```

Comment:
It's not a good idea to use the `index` as a component key. Is there anything else that could be used? `source` perhaps?